src/dft/CMakeLists.txt

```diff
@@ -0,0 +1,279 @@
+  list(GET LISTTYPEID ${T} ID)                              # ID is 1
+  target_compile_definitions(${G} PRIVATE BASETYPEID=${ID} ${COMMON_TARGET_DEFINITIONS})
+endforeach()
+
+# Copy unroll0.org to ${CMAKE_CURRENT_BINARY_DIR}
+
+add_custom_target(unroll0.org_source
+  COMMAND ${CMAKE_EXE_PATH} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/unroll0.org ${CMAKE_CURRENT_BINARY_DIR}
+  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/unroll0.org)
+
+# Target unroll*.c
+
+foreach(T ${LIST_SUPPORTED_FPTYPE})
+  list(GET LISTSHORTTYPENAME ${T} ST)                       # ST is "dp", for example
+  string(TOUPPER ${ST} CST)                                 # CST is "DP"
+  list(GET LISTLONGTYPENAME ${T} LT)                        # LT is "double"
+
+  foreach(E ${ISALIST_${CST}})                              # E is "sse2dp"
+    foreach(N 0 1 2 3)
+      string(CONCAT UC unroll_ ${N} _ ${E} ".c")            # UC is "unroll_0_sse2dp.c"
+      set(UNROLL_TARGET_${CST} ${UNROLL_TARGET_${CST}} ${UC})
+      string(CONCAT UCG ${UC} "_generated")                 # UCG is "unroll_0_sse2dp.c_generated"
+      add_custom_target(${UCG} SOURCES ${UC} ${CMAKE_CURRENT_SOURCE_DIR}/unroll0.org)
+    endforeach()
+  endforeach()
+  message(STATUS "Unroll target for ${CST} : ${UNROLL_TARGET_${CST}}")
+
+  add_custom_command(OUTPUT ${UNROLL_TARGET_${CST}}
+    COMMENT "Generating ${UNROLL_TARGET_${CST}}"
+    COMMAND ${TARGET_MKUNROLL} ${LT} ${ISALIST_${CST}}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    DEPENDS ${TARGET_MKUNROLL} unroll0.org_source
+    )
+endforeach()
+
+# Target unroll*.o
+
+foreach(T ${LIST_SUPPORTED_FPTYPE})
+  list(GET LISTSHORTTYPENAME ${T} ST)                       # ST is "dp", for example
+  string(TOUPPER ${ST} CST)                                 # CST is "DP"
+  list(GET LISTLONGTYPENAME ${T} LT)                        # LT is "double"
+
+  foreach(E ${ISALIST_${CST}})                              # E is "sse2dp"
+    foreach(N 0 1 2 3)
+      string(CONCAT U unroll_ ${N} _ ${E})                  # U is "unroll_0_sse2dp"
+      string(CONCAT UG ${U} "_obj")                         # UG is "unroll_0_sse2dp_obj"
+      string(CONCAT UC ${U} ".c")                           # UC is "unroll_0_sse2dp.c"
+      string(CONCAT UCG ${UC} "_generated")                 # UCG is "unroll_0_sse2dp.c_generated"
+      add_library(${UG} OBJECT ${CMAKE_CURRENT_BINARY_DIR}/${UC} ${CMAKE_BINARY_DIR}/include/sleef.h)
+      set_target_properties(${UG} PROPERTIES ${COMMON_TARGET_PROPERTIES})
+      target_include_directories(${UG} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+      target_compile_definitions(${UG} PRIVATE ${COMMON_TARGET_DEFINITIONS} ${MACRODEF_${E}})
+      target_compile_options(${UG} PRIVATE ${CFLAGS_${E}})
+      add_dependencies(${UG} ${TARGET_HEADERS} ${UCG})
+    endforeach()
+  endforeach()
+endforeach()
+
+# Target libdft
+
+add_library(${TARGET_LIBDFT} SHARED $<TARGET_OBJECTS:dftcommon_obj> $<TARGET_OBJECTS:common_obj> $<TARGET_OBJECTS:arraymap_obj>)
```

Comment:
Instead of using `$<TARGET_OBJECTS:common_obj>`, couldn't you just link the original static library in the next line? Similarly, for `$<TARGET_OBJECTS:arraymap_obj>`: couldn't you have build it as a static library and linked it once in the next line? Am I missing something?